parachain/pallets/commitments/src/lib.rs

```diff
@@ -150,17 +157,21 @@ impl<T: Config> Module<T> {
 impl<T: Config> MessageCommitment for Module<T> {
 
 	// Add a message for eventual inclusion in a commitment
-	// TODO (Security): Limit number of messages per commitment
-	//   https://github.com/Snowfork/polkadot-ethereum/issues/226
 	fn add(channel_id: ChannelId, target: H160, nonce: u64, payload: &[u8]) -> DispatchResult {
-		MessageQueues::append(
```

Comment:
We do need to use [StorageValue::append](https://substrate.dev/rustdocs/v3.0.0/frame_support/storage/types/struct.StorageValue.html#method.append). Its an optimization that allows us to append an item without having to read and decode the entire vector.

I suggest ditching `try_mutate`, and instead use [StorageValue::decode_len](https://substrate.dev/rustdocs/v3.0.0/frame_support/storage/types/struct.StorageValue.html#method.decode_len) to retrieve the length of the vector (its also optimized, and is able read the length without decoding the entire vector).

Reply:
Nice. I'll update